### DISCUSSION TITLE
Java 16 + Postgresql

### DISCUSSION THREAD

**Discussion by alantonica** [vladimir-bukhtoyarov/bucket4j#348] [Q&A] [answered]
Hi,

I'm trying to add bucket4j on my project (spring boot) which uses a postgresql db. There are a couple of questions I want to ask:

1. I see we can override the table name in which the library is looking, but when or where can i do that in a spring boot app?
    Do we create a bean of `PostgreSQLSelectForUpdateBasedProxyManager proxyManager` ?

 Edit: I think i figured this out. We can create a bean with `PostgreSQLSelectForUpdateBasedProxyManager proxyManager` that uses a constant `SQLProxyConfiguration` which will reference the custom table etc.

2. I've created the `buckets` table as per documentation, but the moment I add the 
```
    <dependency>
      <groupId>com.bucket4j</groupId>
      <artifactId>bucket4j-postgresql</artifactId>
      <version>8.2.RC2</version>
    </dependency>
```
to my project, it stops working. 

```
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make 
protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
 throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" 
to unnamed module @35ef1869
```

Thank you. If you need me to provide more information, just let me know.

```
Spring boot 2.6.3
```

```
openjdk 16.0.1 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9)
OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)
```

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
Hello @alantonica 

Most likely problem is caused by jarhell related to Hikari datasource. Unfortunately it has been defined in [bucket4j-postgresql pom.xml](https://github.com/bucket4j/bucket4j/blob/master/bucket4j-postgresql/pom.xml#L53) with wrong scope. Legacy com.zaxxer:HikariCP:java6:2.3.8 is not compatible with JPMS.

Try to fix dependency in this way:
```xml
       <dependency>
          <groupId>com.bucket4j</groupId>
          <artifactId>bucket4j-postgresql</artifactId>
          <version>8.2.RC2</version>
          <exclusions>
              <exclusion>
                  <groupId>com.zaxxer</groupId>
                  <artifactId>HikariCP-java6</artifactId>
              </exclusion>
          </exclusions>
      </dependency>
```

Anyway, pom.xml for bucket4j-postgresql as well as bucket4j-mysql will be fixed in next release.

**Reply by alantonica** [reply]
Thank you. It worked 👍
